main.tf

```diff
@@ -13,6 +13,15 @@ locals {
   default_ocp_version = "${data.ibm_container_cluster_versions.cluster_versions.default_openshift_version}_openshift"
   ocp_version         = var.ocp_version == null || var.ocp_version == "default" ? local.default_ocp_version : "${var.ocp_version}_openshift"
 
+  # To verify rhcos operating system only for OCP versions >=4.15
+  # tflint-ignore: terraform_unused_declarations
+  rhcos_validation = var.operating_system == null || var.operating_system == "REDHAT_8_64" || (var.operating_system == "RHCOS" && (var.ocp_version != null ? var.ocp_version >= "4.15" : false)) ? true : tobool("RHCOS requires VPC clusters created from 4.15 onwards. Upgraded clusters from 4.14 cannot use RHCOS")
```

Comment:
I don't think your logic is taking into consideration that:
- var.ocp_version can be `null`, `"default"`, or an actual value (e.g. `"4.15"`)
- that the default value will change over time. Currently its 4.15, but that will be to the next version in time

Reply:
Made the changes suggested above.